client/castApp/src/castinit.c

```diff
@@ -65,13 +67,115 @@ static void casthook(initHookState state)
     epicsAtExit(&castexit, NULL);
 }
 
+/*
+    Example call: addReccasterEnvVars("SECTOR") or addReccasterEnvVars("SECTOR", "BUILDING")
+    Appends the given env variables to the extra_envs list to be sent in addition to the default_envs array
+*/
+static void addReccasterEnvVars(caster_t* self, int argc, char **argv)
+{
+    int i, j;
+    int ret = 0;
+    if(argc < 2) {
+        errlogSevPrintf(errlogMinor, "At least one argument expected for addReccasterEnvVars\n");
+        return;
+    }
+    /* sanitize input - check for dups and empty args. First arg in argv is function name so skip that */
+    int argCount = argc - 1;
+    epicsMutexMustLock(self->lock);
+    for(i = 1; i < argc; i++) {
+        if(argv[i] == NULL) {
+            argCount--;
+            errlogSevPrintf(errlogMinor, "Arg is NULL for addReccasterEnvVars\n");
+            continue;
+        }
+        else if(argv[i][0] == '\0') {
+            argCount--;
+            errlogSevPrintf(errlogMinor, "Arg is empty for addReccasterEnvVars\n");
+            argv[i] = NULL;
+            continue;
+        }
+        /* check if dup in self->extra_envs. doesn't check if arg is in default_envs right now */
```

Comment:
Are you concerned about duplicates in the argument list as well?

Maybe a simpler approach would be to

1. allocate space for `num_extra_envs+argc` as `new_extra_envs[]`
2. copy in existing `extra_envs[]`
3. for each argv[i], look for duplicates in `new_extra_envs[]`, then either append to skip.

This would avoid the need to look multiple times.